### PR TITLE
Features and fixes around TV station logos

### PR DESCRIPTION
**Pull request by wutschel**
## Description

This PR resolves an issue report from a [forum post](https://forum.kodi.tv/showthread.php?tid=359717&pid=3027965#pid3027965).

1. Monochrome images were not properly analyzed for their average color / brightness causing an inadequate choice of background color. This PR reworks the calculation method.
2. The TV station logs for channel list, recording list and recording details now use same aspect ratios for improved UI consistency.

Screenshots:
https://abload.de/img/simulatorscreenshot-ia2k9f.png (monochrome before, recording + channel list)
https://abload.de/img/simulatorscreenshot-i53jr8.png (monochrome before, recording list + details)
https://abload.de/img/simulatorscreenshot-i96j2c.png (monochrome after, recording + channel list)
https://abload.de/img/simulatorscreenshot-ih3kp4.png (monochrome after, recording list + details)
https://abload.de/img/simulatorscreenshot-ibqjxp.png (RGBA after, recording + channel list)
https://abload.de/img/simulatorscreenshot-iehkbr.png (RGBA after, recording list + details)

## Summary for release notes

Bugfix: Correct handling of monochrome TV station logos with alpha channel
Feature: Align aspect ratio of TV station logos for channel list, recording list and recording details

### CODE REVIEW

**Comment by wutschel** [comment]
@kambala-decapitator, should we do a test build before merging and ask for testing?

File: XBMC Remote/Utilities.m [hunk 1]
```diff
@@ -40,29 +40,67 @@ - (UIColor *)averageColor:(UIImage *)image inverse:(BOOL)inverse{
...
     if (anyNonAlpha) {
         // no alpha channel present
-        for (int row = 0; row < imageHeight; row++) {
-            const UInt8 *rowPtr = rawPixelData + bytesPerRow * row;
-            for (int column = 0; column < imageWidth; column++) {
-                red    += rowPtr[0];
-                green  += rowPtr[1];
-                blue   += rowPtr[2];
-                rowPtr += stride;
+        if (stride == 4) {
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:43]: better use `switch`

File: XBMC Remote/Utilities.m [hunk 2]
```diff
@@ -40,29 +40,67 @@ - (UIColor *)averageColor:(UIImage *)image inverse:(BOOL)inverse{
...
         // weight color with alpha to ignore transparent sections
-        for (int row = 0; row < imageHeight; row++) {
-            const UInt8 *rowPtr = rawPixelData + bytesPerRow * row;
-            for (int column = 0; column < imageWidth; column++) {
-                red    += rowPtr[0] * rowPtr[3];
-                green  += rowPtr[1] * rowPtr[3];
-                blue   += rowPtr[2] * rowPtr[3];
-                alpha  += rowPtr[3];
-                rowPtr += stride;
+        if (stride == 4) {
```
Comment (kambala-decapitator) [inline XBMC Remote/Utilities.m:74]: `switch`

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]
> @kambala-decapitator, should we do a test build before merging and ask for testing?

I don't find it necessary: can always be reverted later if needed

File: XBMC Remote/Utilities.m (hunk 1 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m]: done

File: XBMC Remote/Utilities.m (hunk 2 above)
Comment (wutschel) [reply XBMC Remote/Utilities.m]: done

**Comment by wutschel** [comment]
> I don't find it necessary: can always be reverted later if needed

Fair enough, then let´s go.

**Pull request merged**